### PR TITLE
[cellular] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

To utilize the auto generation feature for API section the package source code must contain the relevant documentation.

Refs #12239

Fixes ENG-1067

# How

This PR moves the current documentation of `Cellular` API - https://docs.expo.io/versions/latest/sdk/cellular - (with minor tweaks and addition) to the package source code.

The changeset includes also small fix for the Interfaces section.

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-cellular/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 - Fix null cellular information on iOS. ([#12710](https://github.com/expo/expo/pull/12710) by [@randomhajile](https://github.com/randomhajile))
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
+- Add TypeScript type to the exported constants. ([#12838](https://github.com/expo/expo/pull/12838) by [@simek](https://github.com/simek))
```
Comment (bbarthec) [inline packages/expo-cellular/CHANGELOG.md:13]: What specific type are you referring to? 🤔

Comment (Simek) [reply packages/expo-cellular/CHANGELOG.md:13]: https://github.com/expo/expo/pull/12838/files#diff-9aac863a23b1c60e6b4cf38ccf0979d621ea4c8dac127f10c0b2e5d78b5d526eL8-L12

Comment (bbarthec) [reply packages/expo-cellular/CHANGELOG.md:13]:
```suggestion
- Add TypeScript types to the exported constants :`allowsVoip`, `carrier`, `isoCountryCode`, `mobileCountryCode` and `mobileNetworkCode`. ([#12838](https://github.com/expo/expo/pull/12838) by [@simek](https://github.com/simek))
```

**Review by bbarthec — APPROVED** [review]
Good work! 👍

**Pull request merged**